### PR TITLE
New fields added in TF core for providers

### PR DESCRIPTION
**Pull request by paultyng**
See https://github.com/hashicorp/terraform/pull/22583

### CODE REVIEW

**Comment by paultyng** [comment]
Good point, the JSON file was a giant single line, so I wasn't entirely sure where to start or if I should add a new case, but happy to. Let me know what makes sense.

**Comment by paultyng** [comment]
Oh nevermind I see, they are built from the .tf files. I don't know that any provider populates these fields yet, but hopefully soon.

**Comment by paultyng** [comment]
@vancluever @chrisarcand do you have thoughts on fixtures if nothing is using this functionality (yet)? Should I just hand code one? I have to dump the JSON using TF from master (0.13) currently to use it.

File: schemas.go [hunk 1]
```diff
@@ -90,6 +90,14 @@ type SchemaBlock struct {
 
 	// Any nested blocks within this particular block.
 	NestedBlocks map[string]*SchemaBlockType `json:"block_types,omitempty"`
+
+	// The description for this block and format of the description. If
+	// no kind is provided, it can be assumed to be plain text.
+	Description     string `json:"description,omitempty"`
+	DescriptionKind string `json:"description_kind,omitempty"`
```
Comment (vancluever) [inline schemas.go:97]: Is this actually an enum? Maybe we should type it out as such?

**Comment by paultyng** [comment]
It is an enum, added it.

For timeline, this is surfaced to providers to fill in the v2 SDK, which should be releasing in a month or so (planned RC next week I think). So hopefully we will fill in AWS (among other providers) soon after that.

File: schemas.go [hunk 2]
```diff
@@ -83,13 +83,28 @@ type Schema struct {
 	Block *SchemaBlock `json:"block,omitempty"`
 }
 
+type StringKind string
```
Comment (vancluever) [inline schemas.go:87]: A bikeshed, but maybe we could think of a more descriptive name for this type? `tfjson.StringKind` is not entirely too descriptive of the purpose here. Also, and a doc comment be added?

Comment (paultyng) [reply schemas.go:87]: Reasonable, I went with a generic name as that is the type name in the protocol: https://github.com/hashicorp/terraform/blob/master/docs/plugin-protocol/tfplugin5.2.proto#L73

But I can make it `DescriptionKind` or something if you prefer, just not sure it will be limited to Description's going forward.

Comment (vancluever) [reply schemas.go:87]: I'm for longer types, even if it's a bit wordy 😛 

Maybe `SchemaBlockDescriptionKind`? 😆

Comment (paultyng) [reply schemas.go:87]: Its used in attribute too, i'll go with `SchemaDescriptionKind`

File: schemas.go [hunk 3]
```diff
@@ -83,13 +83,33 @@ type Schema struct {
 	Block *SchemaBlock `json:"block,omitempty"`
 }
 
+// SchemaDescriptionKind is the format of a string
```
Comment (vancluever) [inline schemas.go:86]:
```suggestion
// SchemaDescriptionKind describes the format type for a particular description's field.
```
